### ISSUE TITLE
Docker Desktop for Windows 10 Home stuck on starting

### ISSUE THREAD

**Issue by ActiveChooN** [docker/for-win#8308]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: F38E54F8-835F-4529-9DCE-97B9AA704721/20200901190245

### Expected behavior

Docker should start normally

### Actual behavior

Docker desktop failed to start and stuck on `starting` status

### Information

  - Windows Version: Windows 10 Home 2004 19041.450
  - Docker Desktop Version: 2.3.6.0
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No, just docker for wsl2

### Steps to reproduce the behavior

  1. Just start docker-desktop
 
Also tried to reboot machine, reinstall docker-desktop and unregister and reinstall wsl docker-desktop distribution

**Comment by KristianI** [comment]
Same issue here.

Except that I got it to start by running `wsl --update` in powershell followed by `wsl --shutdown` 

But there is still no connection from wsl to docker. Also, from the docker dashboard the containers cannot be started.

**Comment by msft-mw** [comment]
I'm also seeing this, but `wsl --update` doesn't work; the `--update` flag doesn't appear to be supported. When I try it, I just get the help list of commands

**Comment by simonferquel** [comment]
@KristianI and @msft-mw can you send diagnostics ?
@ActiveChooN for some reason dockerd within our wsl distro fails to start... can you try to reset to factory defaults and see if this fix the issue for you ?

**Comment by ActiveChooN** [comment]
@simonferquel, thanks for the quick answer. As i understand it will remove all data including volumes. Is there any way to save them?

**Comment by simonferquel** [comment]
@ActiveChooN you can first try to:
- stop Docker Desktop
- stop WSL (`wsl --shutdown`)
- unregister the docker-desktop distro (which contains binaries, but no data) with `wsl --unregister docker-desktop`
- restart Docker Desktop

It might work if something is corrupted there, and as soon as docker-desktop-data keeps there, you are good to go.
If not, and there is no sensitive data in your volume / container configs (passwords, secrets etc.), you can export it with `wsl --export docker-desktop-data docker-desktop-data-backup.tar` and send it to me via simon.ferquel@docker.com so I can try to reproduce the issue on debug builds.

**Comment by msft-mw** [comment]
Hi @simonferquel, uploaded diagnostics with id: `E6601715-2FDD-4849-96CC-6D0906FB1CBB/20200902180409`

Call stack:

```csharp
System.TimeoutException:
Timed out while wait for Docker service to be ready
   at Docker.LinuxkitDaemonStartup.<StartAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\LinuxkitDaemonStartup.cs:line 43
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__24.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\LinuxWSL2Engine.cs:line 152
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92

```

**Comment by simonferquel** [comment]
@msft-mw 
From the logs:
```
[11:03:15.681][ApiProxy          ][Fatal  ] listen tcp 127.0.0.1:2375: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
[11:03:15.686][ApiProxy          ][Error  ] Process died
```
I see in your settings file that you enabled listening 2375. It would be far better if you did disabled it (for security reason). Otherwise you need to find why Windows forbids you to open the port (can be another process listening on the same port, or a port reservation).

**Comment by msft-mw** [comment]
Listening on 2375 lets WSL talk to Windows Docker, the docker cli is installed inside the WSL Linux environment and then it talks to the Windows Docker runtime.

Unfortunately, I can't seem to get the Settings to load so I can disable it, since it just fails to start. I'll try uninstalling and reinstalling.

**Comment by simonferquel** [comment]
Have you tried to look at port reservation using `netsh int ipv4 show excludedportrange protocol=tcp` ?

**Comment by ActiveChooN** [comment]
@simonferquel, hi again, tried few things but no success. Steps i've tried:
1.  Reset to factory defaults (waited around 15 min but nothing happened and status remained `Docker Desktop is resetting to default...`)
1. `wsl.exe --unregister docker-desktop` and `wsl.exe --unregister docker-desktop-data`
1.  Uninstall Docker Desktop
1. Reboot
1. Install Docker Desktop Edge

But Docker Desktop still in `Starting` status. New diagnostic ID: `F38E54F8-835F-4529-9DCE-97B9AA704721/20200904073332`

**Comment by ActiveChooN** [comment]
Issue resolved with update to docker desktop version 2.4.2.0.
